start/software/container-how-to.md

```diff
@@ -0,0 +1,181 @@
+[title]: - "Creating a Docker Container Image"
+
+[TOC]
+
+This guide is meant to accompany the instructions for using Singularity containers 
+in the Open Science Grid.  You can use your own custom container to run jobs in the 
+Open Science Grid, and we assume that those containers are built using Docker.  This 
+guide describes how to create your own Docker container "image" (the blueprint for 
+the container). Once you have created your custom image, 
+you will need to register the image as described in [this page][osg-containers]
+
+## Install Docker and Get a Docker Hub Account
+
+You'll need a Docker Hub account in order to download Docker and share your 
+Docker container images with the OSG: [DockerHub](https://hub.docker.com/)
+
+Install Docker Desktop to your computer using the appropriate version for your 
+operating system. 
+
+## Identify Components
+
+What software do you want to install? Make sure that you have either the source 
+code or a command that can be used to install it through Linux (like `apt-get` or 
+`yum`). 
+
+You'll also need to choose a "base" container, on which to add your particular 
+software or tools. *We strongly recommend using one of the OSG's published containers 
+as your starting point.* See the available containers on Docker Hub here: 
+[OSG Docker Containers](https://hub.docker.com/u/opensciencegrid)
+The best candidates for you will be containers that have "osgvo" in the name. 
+
+If you prefer, you can base your image on images not already published 
+by OSG, but if you do this, we recommend that as one of the creation steps you 
+create the `/cvmfs` directory. See [Special Cases](#special-cases) below. 
+
+## Build a Container
+
+There are two main methods for generating your own container image. 
+
+1. Editing the `Dockerfile`
+2. Editing the default image using local Docker
+
+We recommend the first option, as it is more reproducible, but the second option 
+can be useful for troubleshooting or especially tricky installs. 
+
+### Editing the `Dockerfile`
+
+Create a folder on your computer and inside it, create a blank text file 
+called `Dockerfile`.  
+
+The first line of this file should include the keyword `FROM` and then 
+the name of a Docker image (from Docker Hub) you want 
+to use as your starting point. If using the OSG's Ubuntu Xenial image that 
+would look like this: 
+
+	FROM opensciencegrid/osgvo-ubuntu-xenial
+
+Then, for each command you want to run to add libraries or software, use the 
+keyword `RUN` and then the command. Sometimes it makes sense to string 
```

Comment:
When does it make sense to string together commands vs using multiple RUN commands? Perhaps a brief sentence stating the use cases would be helpful.